yaks/lib/yaks/format/json_api.rb

```diff
@@ -38,7 +38,7 @@ def serialize_resource(resource)
       def serialize_links(subresources)
         subresources.each_with_object({}) do |resource, hsh|
           next if resource.null_resource?
-          hsh[resource.type] = serialize_link(resource)
+          hsh[resource.rels.first] = serialize_link(resource)
```

Comment:
That sounds like a bug. You're right assuming it should have a (computed) rel if no explicit rel is provided.

`:favourite_books` would be an illegal rel, since according to RFC5899 they would have to be either a registered type (either in the [IANA](http://www.iana.org/assignments/link-relations/link-relations.xhtml) or [microformats](http://microformats.org/wiki/existing-rel-values) registry), or a fully qualified URI.

JSON-API doesn't confirm to RFC5899 though, so you would want to turn those URIs back into keywords, which is why the forementioned, but not yet implemented, "polymorphic" identifiers would come in handy.